Sources/Common/Extensions/StringExtensions.swift

```diff
@@ -71,4 +71,34 @@ public extension String {
         let start = index(startIndex, offsetBy: max(0, range.lowerBound))
         return String(self[start...])
     }
+
+    func getRegion() -> Region {
+        switch self {
+        case "us":
+            return Region.US
+        case "eu":
+            return Region.EU
+        default:
+            return Region.US
+        }
+    }
+
+    func getCioLogLevel() -> CioLogLevel {
+        switch self {
+        case "none":
+            return .none
+        case "error":
+            return .error
+        case "info":
+            return .info
+        case "debug":
+            return .debug
+        default:
+            return .error
+        }
+    }
+
+    func getEvent() -> Metric? {
+        Metric(rawValue: lowercased())
+    }
```

Comment:
Instead of these functions being in this file, I think they should exist in the file where the data type exists. 

`func getRegion() -> Region` goes inside of file `Region` is. 
`func getCioLogLevel() -> CioLogLevel` goes inside file `CioLogLevel` is
etc

Since the file that contains the data type has all of the logic for that data type, I think it makes the most sense for extension files to exist in that file.

Reply:
Also, can you add unit tests for each of these functions?

Reply:
hmm, I don't think `string extensions` should be part of the `Model` of the class. I would change these extensions to their respective models then.

Reply:
I like the changes you made with where the functions are located. 

Could you add some automated tests for these functions added?